### PR TITLE
use a limited amount of Python versions for PR runs

### PR DESCRIPTION
**Pull request by beniwohli**
## What does this pull request do?

the CI runs on PRs are getting prohibitively long. To work around
this, we can use only a limited set of Python versions for PR runs
and do a full, nightly run (the same approach we already use for
framework versions).

### CODE REVIEW

File: .ci/.jenkins_python_full.yml [hunk 1]
```diff
@@ -0,0 +1,3 @@
+PYTHON_VERSION:
```
Comment (v1v) [inline .ci/.jenkins_python_full.yml:1]: should this file contain the content of the `.ci/.jenkins_python.yml` file?

Comment (beniwohli) [reply .ci/.jenkins_python_full.yml:1]: right :) fixed in a3d1158

**Review by v1v — APPROVED** [review]
💯 neat!

File: Jenkinsfile [hunk 2]
```diff
@@ -117,15 +117,17 @@ pipeline {
               dir("${BASE_DIR}"){
                 script {
                   // To enable the full test matrix upon GitHub PR comments
+                  def pythonFile = '.ci/jenkins_python.yml'
```
Comment (v1v) [inline Jenkinsfile:120]:
```suggestion
                  def pythonFile = '.ci/.jenkins_python.yml'
```

File: Jenkinsfile [hunk 3]
```diff
@@ -117,15 +117,17 @@ pipeline {
...
                   def frameworkFile = '.ci/.jenkins_framework.yml'
                   if (env.GITHUB_COMMENT?.contains('full')) {
                     log(level: 'INFO', text: 'Full test matrix has been enabled.')
                     frameworkFile = '.ci/.jenkins_framework_full.yml'
+                    pythonFile = '.ci/jenkins_python_full.yml'
```
Comment (v1v) [inline Jenkinsfile:125]:
```suggestion
                    pythonFile = '.ci/.jenkins_python_full.yml'
```

**Review by basepi — APPROVED** [review]

**Pull request merged**